### PR TITLE
feat(views): UI to configure `dendron.yml`

### PR DESCRIPTION
**Pull request by Harshita-mindfire**
This PR aims to create a UI to configure dendron.yml
__BREAKING CHANGE(dependency update)__: updated antd version to 4.21
vsix: https://github.com/dendronhq/dendron/actions/runs/2845068689
Development-Docs: https://github.com/dendronhq/dendron-docs/pull/31

# Dendron Extended PR Checklist

- NOTE: the links don't work. you'll need to go into the wiki and use lookup to find the note until we fix some issues in the markdown export

### Basics

- [x] code should follow [Code Conventions](dev.process.code)
- [x] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [Avoiding Circular Dependencies](dev.process.code.best-practices).
- [x] sticking to existing conventions instead of creating new ones
  - eg: [if configuration for utilities are already in one module or package, add future utilities there as well](https://github.com/dendronhq/dendron/pull/1960#discussion_r786228021)

### Extended
- General
  - [x] check whether code be simplified
  - [x] check if similar function already exist in the codebase. if so, can it be re-used?
  - [x] check if this change adversely impact performance
- Operations
  - [x] when shipping this change, will it just work or will it introduce additional operational overhead due to complicated interface or known bugs?
- Architecture
  - [~] check if code is introducing changes on a foundational class or interface. if so, call for design review if needed
    - eg: [making changes to DNode](https://github.com/dendronhq/dendron/pull/2158#pullrequestreview-854689586)

### Basics
- [~] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### Extended
- [~] can we track the performance of this change to know if it is _successful_?
  - eg: [see usage for export pod](https://github.com/dendronhq/dendron/pull/2190#pullrequestreview-855715612)

### Basics

- [ ] [Write Tests](dev.process.qa.test) 
- [ ] [Confirm existing tests pass](dev.process.qa.test)
- [x] [Confirm manual testing](dev.process.qa.test) 
- [ ] Common cases tested
- [ ] 1-2 Edge cases tested
- [ ] If your tests changes an existing snapshot, snapshots have been [updated](dev.process.qa.test)

## Docs
- [ ] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
- [x] does this change introduce a new or better way of doing things that others need to be aware of? if so, an async should be created and a process added in [Development](dev) or [Packages](pkg)

### Extended
- [x]  is this a developer BREAKING change? if another person cloning from this branch will need to adjust their dependencies or mental model of the architecture, then it is. if this is the case, make sure this is communicated according to [Close Loop](dev.process.close-loop)
  - eg: [breaking dev change due to new dependency](https://github.com/dendronhq/dendron/pull/2188#pullrequestreview-855696330)

### CODE REVIEW

File: packages/dendron-plugin-views/src/components/DendronConfigure.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+                <Space
+                  direction="vertical"
+                  size="middle"
+                  style={{ display: "flex" }}
+                >
+                  {Object.keys(dendronConfig)
+                    .filter((conf) =>
+                      _.lowerCase(conf).includes(_.lowerCase(searchString))
+                    )
+                    .map((conf) => {
```
Comment (jonathanyeung) [inline packages/dendron-plugin-views/src/components/DendronConfigure.tsx:136]: could we do some manual rearranging of the order of the setting groups? 

I'm thinking maybe
1. Commands
2. Workspace
3. Preview
4. Publishing
5. Global
6. Dev

Current:
image

**Review by jonathanyeung — CHANGES_REQUESTED** [review]
Looking good!

After merging, can you create an async with the https://github.com/dendronhq/dendron-docs/pull/31 information?

**Comment by jonathanyeung** [comment]
@Harshita-mindfire - looks like there's a build issue: 

```log
  "src/utils/ExtensionUtils.ts(71,19): error TS2304: Cannot find name 'getDWorkspace'.\n" +
```

**Review by hikchoi — APPROVED** [review]
Looks great!

LGTM

Some tweaks I would like here, but we can leave them as a next step:
1. Hide unnecessary config keys that the users will likely never want to change manually(e.g. `version`)
2. Make abbreviations / tone of sentences consistent across all config entry descriptions
3. Explicitly mark default empty / required / optional keys accordingly

Some ehancement ideas:
1. Add hover to dropdown selections with additional information on what they do
2. per-config _revert to default_ button
3. highlight filtered items when using the search bar

File: packages/dendron-plugin-views/src/components/DendronConfigure.tsx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+                  height: "100%",
+                  overflowY: "scroll",
+                }}
+              >
+                <Space
+                  direction="vertical"
+                  size="middle"
+                  style={{ display: "flex" }}
+                >
+                  {Object.keys(dendronConfig)
```
Comment (kevinslin) [inline packages/dendron-plugin-views/src/components/DendronConfigure.tsx:132]: can we have two levels of grouping if it exists? (eg. `commands.lookup.note` should be in one box) https://www.loom.com/i/ffc413e6c6f04cd3af9d1af329aa0ebf

**Review by kevinslin — CHANGES_REQUESTED** [review]
nice job! some comments about configuration and display

File: packages/dendron-plugin-views/src/components/DendronConfigure.tsx (hunk 2 above)
Comment (Harshita-mindfire) [reply packages/dendron-plugin-views/src/components/DendronConfigure.tsx]: @kevin, can we do this in a follow-up PR? might need to add more than just a tweak to get this work.

**Review by bpathakota — APPROVED** [review]
PR title looks good

**Comment by kevinslin** [comment]
@Harshita-mindfire no rush - can you ping me when this is ready to re-review with the sub-grouped configs?

**Comment by Harshita-mindfire** [comment]
> @Harshita-mindfire no rush - can you ping me when this is ready to re-review with the sub-grouped configs?

image
@kevinslin ready for re-review

**Review by kevinslin — CHANGES_REQUESTED** [review]
- can we remove the following configuration (for the ui, we should just expose 'high level' configuration)

- fuzzThreshold
- `workspace.dendronVersion`
- `workspace.seeds`
- `workspace.workspace`
- `workspace.disableTelemetry` (this is a global parameter, we are removing the workspace specific one)
- Workspace > Max Previews Cached
- Workspace > Max Note Length
- Workspace > Feedback
- Workspace > Api Endpoint
- Workspace > Workspace Vault Sync Mode
- Publishing > Write Stubs
- Publishing > Segment Key
- Publishing > Cognito User Pool Id
- Publishing > Cognito Client Id
- Publishing > Site Banner
- Publishing > Duplicate Note Behavior
- Dev > Enable Preview V 2
- Dev > Next Server Url
- Dev > Next Static Root
- Dev > Engine Server Port
- Dev > Force Watcher Type

**Comment by Harshita-mindfire** [comment]
> * can we remove the following configuration (for the ui, we should just expose 'high level' configuration)

@kevinslin updated!

**Review by kevinslin — APPROVED** [review]

**Pull request merged**